### PR TITLE
fix(input-selection): roundRobinSelection now ensures all change bundles meet minRequiredAda

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

The was a bug in the `coalesceChangeBundlesForMinCoinRequirement` function used internally by `roundRobinSelection`. This function must ensure that all change bundles meet the minAdaRequired. The bundles would be sorted by coin value and then it would iterate in ascending order until it found bundle that meet the requirement, this meant that if there was a bungle with low coin value that meet the requirement and a bundle with higher coin value that didn't it would go undetected and the bundle would not be coalesced.

# Proposed Solution

Fix the `coalesceChangeBundlesForMinCoinRequirement` function to make sure all bundles are verified and coalesced if needed.

# Important Changes Introduced

- The coalesceChangeBundlesForMinCoinRequirement function should now verify all bundles.
- Added several unit test to the coalesceChangeBundlesForMinCoinRequirement function.

### CODE REVIEW

File: packages/input-selection/src/RoundRobinRandomImprove/change.ts [hunk 1]
```diff
@@ -207,7 +207,21 @@ const pickExtraRandomUtxo = (
   return { utxoRemaining: newUtxoRemaining, utxoSelected: newUtxoSelected };
 };
 
-const coalesceChangeBundlesForMinCoinRequirement = (
+const mergeWithSmallestBundle = (values: Cardano.Value[], index: number): Cardano.Value[] => {
+  const toBeMerged = values.splice(index, 1)[0];
```
Comment (mkazlauskas) [inline packages/input-selection/src/RoundRobinRandomImprove/change.ts:211]: This mixes up the styles: since we're returning the array, we shouldn't mutate the argument.
```suggestion
  const toBeMerged = [...values].splice(index, 1)[0];
```

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

**Review by mkazlauskas — APPROVED** [review]
Fantastic detective work! :detective:

File: packages/input-selection/test/change.test.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    const changeBundles: Cardano.Value[] = [
+      { assets: new Map([[TOKEN1_ASSET_ID, 2333n]]), coins: 2_000_000n },
+      { assets: new Map([[TOKEN2_ASSET_ID, 2333n]]), coins: 2_000_000n },
+      { assets: new Map([[TOKEN3_ASSET_ID, 2333n]]), coins: 2_000_000n }
+    ];
+
+    const result = coalesceChangeBundlesForMinCoinRequirement(changeBundles, computeMinimumCoinQuantity);
+
+    expect(result).toBeUndefined();
+  });
```
Comment (mirceahasegan) [inline packages/input-selection/test/change.test.ts:96]: Great work, @AngelCastilloB . 
I would propose adding a test for empty change list?

Comment (mirceahasegan) [reply packages/input-selection/test/change.test.ts:104]: added test here: [734f6a8](https://github.com/input-output-hk/cardano-js-sdk/pull/633/commits/734f6a8b978be63ad8889a7bec2bad3ab7138bd4)

**Review by mirceahasegan — DISMISSED** [review]

**Review by rhyslbw — APPROVED** [review]
Ready for squashing

**Review by mirceahasegan — APPROVED** [review]
I'll double on "fantastic detective work!" 🕵️  🚀

**Review by rhyslbw — APPROVED** [review]
Great work @AngelCastilloB :bug:

**Review by mkazlauskas — APPROVED** [review]
